drivers/power/ltc7841/iio_ltc7841.c

```diff
@@ -0,0 +1,428 @@
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int ltc7841_iio_write_vout(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel,
+				  intptr_t priv)
+{
+	struct ltc7841_iio_desc *iio_ltc7841 = dev;
+	struct ltc7841_desc *ltc7841 = iio_ltc7841->ltc7841_desc;
+	int32_t val1, val2;
+	double temp_value;
+	uint16_t vout_value;
+
+	iio_parse_value(buf, IIO_VAL_INT_PLUS_MICRO, &val1, &val2);
+	/* value is now in millivolts */
+	temp_value = (double)(val1 * (int32_t)MILLI + (val2 /
+			      MILLI));
+	/* for integer scaling*/
+	temp_value = temp_value * CENTI / LTC7841_MAX_VOUT;
+	temp_value = temp_value * MFR_MAX_VALUE ;
+	/* final percentage value to be written to register, scaling done on the last part*/
+	vout_value = (uint16_t)(temp_value /
+				CENTI);
+	return ltc7841_change_vout_command(ltc7841, vout_value);
+}
+
+/**
+ * @brief Handles the write request for vout-related attributes.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int ltc7841_iio_clear_status(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+	struct ltc7841_iio_desc *iio_ltc7841 = dev;
+	struct ltc7841_desc *ltc7841 = iio_ltc7841->ltc7841_desc;
+	return ltc7841_clear_status_word_bits(ltc7841);
+}
+
+/**
+ * @brief Handles the read request for vout-related attributes.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int ltc7841_iio_read_vout(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct ltc7841_iio_desc *iio_ltc7841 = dev;
+	struct ltc7841_desc *ltc7841 = iio_ltc7841->ltc7841_desc;
+	int ret, temp;
+	int32_t vals[2];
+	uint8_t read_value[2];
+
+	if (!dev)
+		return -EINVAL;
+
+	if (!iio_ltc7841->ltc7841_desc)
+		return -EINVAL;
+
+	ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_VOUT, read_value);
+	if (ret)
+		return ret;
+	temp = no_os_get_unaligned_le16(read_value);
+	vals[0] = (int32_t)(temp / CENTI);
+	vals[1] = (int32_t)(temp % CENTI) * DECIMAL_POINT_SCALE_VOLTAGE;
+
+	return iio_format_value(buf, len, IIO_VAL_INT_PLUS_MICRO, 1, vals);
+}
+
+static struct iio_attribute ltc7841_chan_attrs[] = {
+	{
+		.name = "raw",
+		.show = ltc7841_iio_read_raw
+	},
+	END_ATTRIBUTES_ARRAY
+};
```

Comment:
Is the scale of every channel equal to 1? Otherwise, you also need the scale attribute here.

Reply:
The scale of every channel is the same at 10mV per bit.

Reply:
on my next commit, I'll add as you stated below a scale attribute for the channel

Reply:
When implementing review suggestions, you don't need to create a new commit (unless you add new functionality, which should have its own commit). Just modify the current one and push the changes again. After that, you should add a changelog comment in this PR stating what you did since the last push.